### PR TITLE
cosmomc: Enable Python bindings

### PR DESCRIPTION
**Pull request by eschnett**

### CODE REVIEW

File: var/spack/repos/builtin/packages/cosmomc/package.py [hunk 1]
```diff
@@ -51,12 +51,19 @@ def url_for_version(self, version):
...
 
     depends_on('mpi', when='+mpi')
     depends_on('planck-likelihood', when='+planck')
-    depends_on('python@2.7:2.8,3.4:')
+    depends_on('py-matplotlib')
+    depends_on('py-numpy')
+    depends_on('py-pandas')
+    depends_on('py-scipy')
+    depends_on('py-six')
+    depends_on('python @2.7:2.999,3.4:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/cosmomc/package.py]: These should all be `type=('build', 'run')`.

**Comment by citibeth** [comment]
I looked a bit upstream.  I agree with @adamjstewart, the Python bindings should almost certainly be optional.  Use this for an optional extends:
```
    extends('python', when='+python')
```

File: var/spack/repos/builtin/packages/cosmomc/package.py (hunk 1 above)
Comment (eschnett) [inline var/spack/repos/builtin/packages/cosmomc/package.py]: Done.

**Comment by eschnett** [comment]
Converted Python dependency into a variant.

**Pull request merged**